planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -808,6 +820,41 @@ void planning_scene_monitor::PlanningSceneMonitor::currentWorldObjectUpdateCallb
     }
 }
 
+void planning_scene_monitor::PlanningSceneMonitor::syncSceneUpdates(const ros::Time &t)
+{
+  if (t.isZero())
+    return;
+
+  ROS_DEBUG_STREAM_NAMED("PSM", "sync to: " << fmod(t.toSec(), 10.));
+  enforce_next_state_update_ = true;  // enforce next state update to trigger without throttling
+
+  // Robot state updates in the scene are only triggered by the state monitor on changes of the state.
+  // Hence, last_state_update_time_ might be much older than current_state_monitor_ (when robot didn't moved for a while).
+  boost::shared_lock<boost::shared_mutex> lock(scene_update_mutex_);
+  ros::Time last_robot_update = current_state_monitor_ ? current_state_monitor_->getCurrentStateTime() : ros::Time();
+  while (current_state_monitor_ &&                     // sanity check
+         last_robot_update < t &&                      // wait for recent state update
+         (t - last_robot_motion_time_).toSec() < 1.0)  // but only if robot moved in last second
```

Comment:
It's highly important! I was hoping, the comment explains it. I'll try again:

This was only relevant on rviz, which uses a PSM without a robot state monitor. Hence rviz receives it's scene updates via planning scene diffs from `move_group`'s PSM that actually monitors the robot state. However, publishing scene diffs is [throttled in PSM](https://github.com/ros-planning/moveit_ros/blob/9596574fcabd8dd4c6e0603a054138f8b2ec924a/planning/planning_scene_monitor/src/planning_scene_monitor.cpp#L382), which means that rviz' PSM should wait for a while until move_group's PSM has definitely published an update.

Actually one second seems even too short, as the default is [`2s`](https://github.com/ros-planning/moveit_ros/blob/9596574fcabd8dd4c6e0603a054138f8b2ec924a/planning/planning_scene_monitor/src/planning_scene_monitor.cpp#L220). The actual value isn't really important. As soon as an update is received, the conditioned wait finishes.